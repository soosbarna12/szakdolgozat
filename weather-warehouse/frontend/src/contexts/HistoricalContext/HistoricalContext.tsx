import React, { createContext, useEffect, useState } from "react";
import { LOCAL_STORAGE_TEMPERATURE_SCALE } from "../../consts/temperatureScale.const";
import { TemperatureScale } from "../../types/temperatureScale.type";
import { HistoricalContextType, HistoricalLocationData, HistoricalPageData, HistoricalProviderProps } from "./HistoricalContext.type";


export const HistoricalContext = createContext<HistoricalContextType>({
  location: { name: "", lat: 0, lon: 0 },
  setLocation: () => { },
  historicalPageData: [] as any,
  setHistoricalPageData: () => { },
  temperatureScale: TemperatureScale.Celsius,
  setTemperatureScale: () => { },
});

export const HistoricalLocationProvider = ({ children }: HistoricalProviderProps) => {

  const [location, setLocation] = useState<HistoricalLocationData>(() => {
    const storedLocation = localStorage.getItem("historicalLocation");
    const defaultLocation = { name: "", lat: 0, lon: 0 };
    try {
      return storedLocation ? JSON.parse(storedLocation) : defaultLocation;
    } catch (error) {
      console.error("Failed to parse location from localStorage:", error);
      return defaultLocation;
    }
  });

  const [historicalPageData, setHistoricalPageData] = useState<HistoricalPageData[]>(() => {
    const storedData = localStorage.getItem("historicalPageData");
    try {
      return storedData ? JSON.parse(storedData) : [];
    } catch (error) {
      console.error("Failed to parse historical data from localStorage:", error);
      return [];
    }
  });

  const [temperatureScale, setTemperatureScale] = useState<TemperatureScale | string>(() => {
    const storedTemperatureScale = localStorage.getItem(LOCAL_STORAGE_TEMPERATURE_SCALE);
    try {
      return storedTemperatureScale ? JSON.parse(storedTemperatureScale) : TemperatureScale.Celsius;
    } catch (error) {
      console.error("Failed to parse temperature scale from localStorage:", error);
      return TemperatureScale.Celsius;
    }
  });


  useEffect(() => {
    if (historicalPageData) {
      localStorage.setItem("historicalPageData", JSON.stringify(historicalPageData));
    }
  }, [historicalPageData]);

  useEffect(() => {
    localStorage.setItem("historicalLocation", JSON.stringify(location));
  }, [location]);

  useEffect(() => {
    localStorage.setItem(LOCAL_STORAGE_TEMPERATURE_SCALE, JSON.stringify(temperatureScale));
  }, [temperatureScale]);


  return (
    <HistoricalContext.Provider value={{ location, setLocation, historicalPageData, setHistoricalPageData, temperatureScale, setTemperatureScale }}>
      {children}
    </HistoricalContext.Provider>
  );
};
