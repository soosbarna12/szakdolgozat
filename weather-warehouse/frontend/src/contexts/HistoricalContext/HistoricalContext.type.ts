import { Dispatch, ReactNode, SetStateAction } from "react";
import { HistoricalDataTable } from "../../types/historicalDataTable.type";
import { TemperatureScale } from "../../types/temperatureScale.type";

export interface HistoricalLocationData {
  name: string;
  country?: string;
  lat: number;
  lon: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-object-type
export interface HistoricalPageData extends HistoricalDataTable {}

export interface HistoricalContextType {
  location: HistoricalLocationData;
  historicalPageData: any [];
  temperatureScale: TemperatureScale | string;
  setLocation: Dispatch<SetStateAction<HistoricalLocationData>>;
  setHistoricalPageData: Dispatch<SetStateAction<HistoricalPageData[]>>;
  setTemperatureScale: Dispatch<SetStateAction<TemperatureScale | string>>;
}

export interface HistoricalProviderProps {
  children: ReactNode;
}
