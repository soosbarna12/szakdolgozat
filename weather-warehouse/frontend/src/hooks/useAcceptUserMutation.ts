import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useAcceptUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.post(`/user/accept`, { id });
      return response.data;
    },
    onSuccess: () => {
      showAlert("User accepted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to accept user", "error");
    },
  });

  return mutation;
}
