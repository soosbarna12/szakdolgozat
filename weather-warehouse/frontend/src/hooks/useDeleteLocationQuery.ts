import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";

export function useDeleteLocationQuery() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (userLocationID: number) => {
      const response = await axios.delete(`/user/deleteLocation`, { params: { userLocationID } });
      return response.data;
    },
    onSuccess: () => {
      showAlert("Location deleted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to delete location", "error");
    },
  });

  return mutation;
}