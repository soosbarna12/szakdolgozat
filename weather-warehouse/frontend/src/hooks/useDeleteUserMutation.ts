import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useDeleteUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.delete(`/user/delete`, { params: { id } });
      return response.data;
    },
    onSuccess: () => {
      showAlert('User deleted successfully', 'success');
    },
    onError: () => {
      showAlert('Failed to delete user', 'error');
    },
  });

  return mutation;
}
