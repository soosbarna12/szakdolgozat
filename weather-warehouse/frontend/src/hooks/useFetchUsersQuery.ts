import { useQuery } from "@tanstack/react-query";
import { User } from "../types/userData.type";
import axios from "../utils/axiosConfig";


export function useFetchUsersQuery() {
  
  const { data, error, isLoading, refetch } = useQuery<User[]>({
    queryKey: ["users"],
    queryFn: async () => {
      const response = await axios.get(`/api//user/userData`);

      return response.data as User[];
    },
  });

  return { data, error, isLoading, refetch };
}
