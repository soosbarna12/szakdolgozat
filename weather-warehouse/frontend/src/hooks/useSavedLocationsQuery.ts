import { useQuery } from "@tanstack/react-query";
import { useEffect } from "react";
import { SavedUserLocation } from "../types/historicalDataTable.type";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useSavedLocationQuery(open: boolean) {
  const { showAlert } = useAlert();
  
  const { data: savedLocations, error, isLoading, refetch } = useQuery({
    queryKey: ["fetchSavedLocations", open],
    queryFn: async () => {
      const response = await axios.get(`/api/user/fetchSavedLocations`);
      return response.data as SavedUserLocation[];
    },
    enabled: open,
    retry: false,
  });
  
  useEffect(() => {
    if (error) {
      showAlert("Error fetching saved locations", "error");
    }
  }, [error]);

  return { savedLocations, error, isLoading, refetch };
}
