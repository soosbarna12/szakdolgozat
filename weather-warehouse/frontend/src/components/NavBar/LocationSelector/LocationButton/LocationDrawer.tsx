import DeleteIcon from '@mui/icons-material/Delete';
import PlaceOutlinedIcon from '@mui/icons-material/PlaceOutlined';
import { Box, Drawer, IconButton, List, ListItem, ListItemText, Skeleton, Toolbar, Tooltip, Typography } from "@mui/material";
import dayjs from 'dayjs';
import React, { useContext } from "react";
import { HistoricalContext } from '../../../../contexts/HistoricalContext/HistoricalContext';
import { useDeleteLocationQuery } from '../../../../hooks/useDeleteLocationQuery';
import { useSavedLocationQuery } from '../../../../hooks/useSavedLocationsQuery';
import { StyledButton, StyledIconButton } from '../../../../stlyes/button.style';
import { LocationDrawerProps } from "./LocationDrawer.type";


export function LocationDrawer(props: Readonly<LocationDrawerProps>) {
  const { toggleLocationDrawer, open } = props;
  const { savedLocations, isLoading, refetch } = useSavedLocationQuery(open);
  const { mutateAsync: deleteLocation } = useDeleteLocationQuery();
  const { setHistoricalPageData, setLocation } = useContext(HistoricalContext);


  function handleDrawerClose() {
    toggleLocationDrawer(false);
  }

  function handleOpenSavedLocation(index: number) {
    if (!savedLocations) return;
    setHistoricalPageData(savedLocations[index].locationData);
    setLocation({ name: "", lat: 0, lon: 0 });
  }

  function handleDeleteLocation(userLocationID: number, event: React.MouseEvent<HTMLElement>) {
    event.stopPropagation();
    deleteLocation(userLocationID).then(() => { refetch() });
  }

  function renderLocations() {
    if (isLoading) {
      return [0, 1, 2].map((index) => (
        <ListItem key={index} data-testid="locationDrawerSkeleton">
          <Skeleton
            variant="rectangular"
            width="100%"
            height={100}
            sx={{ borderRadius: "20px" }}
          />
        </ListItem>
      ));
    }

    if (savedLocations && savedLocations?.length > 0) {
      return savedLocations.map((arrayElement, index) => {
        return (
          <ListItem key={index} disablePadding onClick={() => handleOpenSavedLocation(index)} data-testid="savedLocation">
            <StyledButton fullWidth variant="text" color="primary" sx={{ height: "100px", boxShadow: 4, margin: 1 }} >
              <ListItemText
                primary={arrayElement?.locationData?.at(-1)?.cityName}
                secondary={
                  <>
                    <span>Filtered Date: {arrayElement?.locationData?.at(-1)?.date}</span>
                    <span>Saved Date: {dayjs(arrayElement.dateSaved).format("YYYY-MM-DD")}</span>
                  </>}
              />
            </StyledButton>
            <IconButton
                edge="end"
                aria-label="delete"
                onClick={(event) => handleDeleteLocation(arrayElement.userLocationID, event)}
                sx={{ marginLeft: 1 }}
                data-testid="deleteButton"
              >
                <DeleteIcon />
              </IconButton>
          </ListItem >
        )
      });
    }

    return <Typography>No saved locations found.</Typography>
  }


  return (
    <Drawer
      open={open}
      onClose={handleDrawerClose}
      elevation={4}
      slotProps={{ backdrop: { style: { opacity: 1 } } }}>
      <Box sx={{ width: 300 }} paddingRight={"10px"} paddingLeft={"24px"} paddingTop={"8px"}>
        <List sx={{ padding: 0 }}>
          <Toolbar disableGutters>
            <Tooltip title="Close locations">
              <StyledIconButton
                onClick={handleDrawerClose}
                sx={{
                  padding: 0, margin: "10px", width: "48px", height: "48px",
                  borderRadius: "100%", boxShadow: 4, outline: 0
                }}>
                <PlaceOutlinedIcon sx={{ padding: 0, margin: 0, width: "32px", height: "32px" }} />
              </StyledIconButton>
            </Tooltip>
            <Typography variant="body1" textTransform={"uppercase"}>Saved locations</Typography>
          </Toolbar>
          {renderLocations()}
        </List>
      </Box >
    </Drawer>
  );
}
