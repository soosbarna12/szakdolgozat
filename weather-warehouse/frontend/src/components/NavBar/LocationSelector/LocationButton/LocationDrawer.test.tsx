import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { HistoricalContext } from '../../../../contexts/HistoricalContext/HistoricalContext';
import { useDeleteLocationQuery } from '../../../../hooks/useDeleteLocationQuery';
import { useSavedLocationQuery } from '../../../../hooks/useSavedLocationsQuery';
import { renderWithQueryClient } from '../../../../utils/test/renderWithQueryClient';
import { LocationDrawer } from './LocationDrawer';

jest.mock('../../../../hooks/useSavedLocationsQuery', () => ({
  useSavedLocationQuery: jest.fn(),
}));

jest.mock('../../../../hooks/useDeleteLocationQuery', () => ({
  useDeleteLocationQuery: jest.fn(),
}));

describe('NavBar/LocationSelector/LocationDrawer', () => {
  const mockToggleLocationDrawer = jest.fn();
  const mockSetHistoricalPageData = jest.fn();
  const mockSetLocation = jest.fn();

  const renderComponent = (open = true) => {
    return renderWithQueryClient(
      <HistoricalContext.Provider
        value={{
          setHistoricalPageData: mockSetHistoricalPageData,
          setLocation: mockSetLocation,
        } as any}
      >
        <LocationDrawer toggleLocationDrawer={mockToggleLocationDrawer} open={open} />
      </HistoricalContext.Provider>
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();

    (useSavedLocationQuery as jest.Mock).mockReturnValue({
      savedLocations: [
        {
          locationData: [{ cityName: 'New York', date: '2023-01-01' }],
          dateSaved: '2023-01-02',
          userLocationID: 1,
        },
      ],
      isLoading: false,
      refetch: jest.fn(),
    });

    (useDeleteLocationQuery as jest.Mock).mockReturnValue({
      mutateAsync: jest.fn(() => Promise.resolve()),
    });
  });

  it('matches LocationDrawer component snapshot', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders saved locations and handles opening a saved location', async () => {
    renderComponent();

    // Verify saved location is rendered
    const savedLocation = screen.getByText('New York');
    expect(savedLocation).toBeInTheDocument();

    // Click on the saved location
    fireEvent.click(savedLocation);

    // Verify setHistoricalPageData and setLocation are called
    await waitFor(() => {
      expect(mockSetHistoricalPageData).toHaveBeenCalledWith([{ cityName: 'New York', date: '2023-01-01' }]);
      expect(mockSetLocation).toHaveBeenCalledWith({ name: '', lat: 0, lon: 0 });
    });
  });

  it('handles deleting a saved location', async () => {
    renderComponent();

    // Click on the delete button
    const deleteButton = screen.getByLabelText('delete');
    fireEvent.click(deleteButton);

    // Verify refetch is called
    await waitFor(() => {
      expect(screen.getByText('New York')).toBeInTheDocument(); // Ensure the location is still rendered after deletion
    });
  });

  it('renders skeletons when loading', () => {
    (useSavedLocationQuery as jest.Mock).mockReturnValueOnce({
      savedLocations: [],
      isLoading: true,
      refetch: jest.fn(),
    });

    renderComponent();

    // Verify skeletons are rendered
    const skeletons = screen.getAllByTestId('locationDrawerSkeleton');
    expect(skeletons).toHaveLength(3);
  });

  it('renders "No saved locations found" when there are no saved locations', () => {
    (useSavedLocationQuery as jest.Mock).mockReturnValueOnce({
      savedLocations: [],
      isLoading: false,
      refetch: jest.fn(),
    });

    renderComponent();

    // Verify "No saved locations found" message is rendered
    expect(screen.getByText('No saved locations found.')).toBeInTheDocument();
  });

  it('closes the drawer when the close button is clicked', () => {
    renderComponent();

    // Click on the close button
    const closeButton = screen.getByLabelText('Close locations');
    fireEvent.click(closeButton);

    // Verify toggleLocationDrawer is called
    expect(mockToggleLocationDrawer).toHaveBeenCalledWith(false);
  });
});