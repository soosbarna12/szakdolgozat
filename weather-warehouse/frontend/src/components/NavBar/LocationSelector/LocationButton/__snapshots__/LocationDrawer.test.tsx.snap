// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar/LocationSelector/LocationDrawer matches LocationDrawer component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-anchorLeft MuiDrawer-modal MuiModal-root css-2soe65-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-4nmryk-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-cyqh1t-MuiPaper-root-MuiDrawer-paper"
        style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
        tabindex="-1"
      >
        <div
          class="MuiBox-root css-u2bmcb"
        >
          <ul
            class="MuiList-root MuiList-padding css-2wvk20-MuiList-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-1u9o2lj-MuiToolbar-root"
            >
              <button
                aria-label="Close locations"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-2hpya9-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gtean2-MuiSvgIcon-root"
                  data-testid="PlaceOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m6-1.8C18 6.57 15.35 4 12 4s-6 2.57-6 6.2c0 2.34 1.95 5.44 6 9.14 4.05-3.7 6-6.8 6-9.14M12 2c4.2 0 8 3.22 8 8.2 0 3.32-2.67 7.25-8 11.8-5.33-4.55-8-8.48-8-11.8C4 5.22 7.8 2 12 2"
                  />
                </svg>
              </button>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ar4n1-MuiTypography-root"
              >
                Saved locations
              </p>
            </div>
            <li
              class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              data-testid="savedLocation"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-lsfikd-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                  >
                    New York
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
                  >
                    <span>
                      Filtered Date: 
                      2023-01-01
                    </span>
                    <span>
                      Saved Date: 
                      2023-01-02
                    </span>
                  </p>
                </div>
              </button>
              <button
                aria-label="delete"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1l3by4e-MuiButtonBase-root-MuiIconButton-root"
                data-testid="deleteButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
